Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYApplePayPaymentProvider.m

```diff
@@ -176,10 +180,20 @@ - (void)proceedWithApplePay
 
 	PKPaymentRequest *request = [self paymentRequest];
 	request.paymentSummaryItems = [self.checkout buy_summaryItemsWithShopName:self.shop.name];
-	PKPaymentAuthorizationViewController *controller = [[PKPaymentAuthorizationViewController alloc] initWithPaymentRequest:request];
+	
+	Class PaymentClass = [self applePayController];
+	id controller = [[PaymentClass alloc] initWithPaymentRequest:request];
 	if (controller) {
-		controller.delegate = self;
-		[self.delegate paymentProvider:self wantsControllerPresented:controller];
+		[controller setDelegate:self];
+		if ([PaymentClass isSubclassOfClass:[PKPaymentAuthorizationController class]]) {
```

Comment:
Can we clean up this logic a bit? There are a lot of conditionals and some ugly introspection and an `id` typed variable.

One approach would be extract the conditional tests into their own methods, and then flatten the conditionals.

We could also extract this all to one method, `-presentPaymentController:`.